### ISSUE TITLE
this issue is to reissue #40

### ISSUE THREAD

**Issue by idontveapc** [namidaco/namida#55]
#40
> i was to reopen my issue (#40), but it can be done by the one who closes the issue/discussion. i didn't close it that's why opening a new.

the issue was as namida worked fine and didn't crash after updating. but the real issue was INDEXER.

* as said "INDEXER keeps running".

why im reopening a new; because:
- when updated from 1.4.8 to 1.5.4 and then to 1.6.8 the app didn't run indexer, until yesterday i **re-index** and **what happened??** IT CRASHED (but after one crash it didn't crash again).
- then to test indexer i renamed(note: *renamed in emoji glyphs and not in letter/text(s)*) a song file. (note: *the file was not in the `/storage/emulated/0/Music directory` instead another directory which namida indexed actually*) namida asks for storage access AGAIN. and app crashes when you allow it then indexer runs and then when you open namida again you can't play the renamed file but the first song on the TRACKS (note: *which is a song file `.mp3/.m4a/.flac`*) is playing the song but showing a video without its original audio!! i believe that video was from my downloads though but audio was from the song file.
- now to fix the video problem, i had to clear the all data.(note: *when i cleared data and when im testing, permission for storage is allowed and for all files*) NOW WHAT?? version is 1.6.8 so i stupid believed it won't crash!? it crashed while indexer kept indexing and it's not a one crash. you can't use namida until indexer won't finish but namida is crashing. constantly.

now i had to go back to the version 1.4.8 and indexer finishes its job, namida doesn't crash. i can update namida now but there are a lot of ptoblems with indexing, and how new files would be added without running indexer again!!?

i think better would be to not extract meta like this, instead read the files and when they change.

**Comment by MSOB7YY** [comment]
hello, sorry for the inconvenience

the recent versions (starting from v1.4.8) implements a new way of indexing (https://github.com/namidaco/namida/commit/5ae0247a62046d03ce9cdf486f908a54ee085de7) which speeds it up by 250% using concurrent reading, this might be causing the crash.
im thinking about adding a flag to adapt the old behavior again, this could fix the issue completely 

- about the second point, it's displaying the old track name/metadata because the indexing wasn't done, thus using the old file with old data, its also not playing audio bcz of the same reason (path changed but not updated in namida)
   - u can update by either a `refresh` or a full `re-indexing`
 
- its weird not being able to use namida while indexing, normally it should happen silently without affecting the experience, unless tags reading fails and switches to `ffmpeg`, this might cause some lag

- TO SUM THINGS UP:
   - if ur music is on sd card, the indexing might be even slower duo to low card read/write speeds
   - as for now, u can use Media Store API, which does not index anything, it just reads some ready values provided by android, so it will be instant
   - im not a fan of "restart ur machine" but "@Joeeeeb5" fixed a similar issue by restarting his phone lmao, u can try too

**Comment by Joeeeeb5** [comment]
You can also try excluding and reselecting the directory

**Comment by MSOB7YY** [comment]
thank you for detailed report, i think the indexing is working now?
- as for video get indexed as audio:
   - it's definitely cuz the file has an audio extensions, check https://github.com/namidaco/namida/blob/6104374aaf318f74226b015aef35f5e286d30f28/lib/core/constants.dart#L179
- as for the miniplayer:
  1. you can swap artist & title places (from customization settings > miniplayer)
  2. currently the waveform resampler tries to keep some sort of normalization among all tracks, if the track has high samples it will cause it to get bigger like this, I'll look into another workaround for that

**Comment by MSOB7YY** [comment]
if u have artworks full res extracted (i.e u disabled media store and did a re-index) then it might be a bit slower to extract colors, but it only happens once.

anyways u restarted and everything is working fine, means i can take this as a "oki u can close issue now"? 😅
